File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/MvcApplication1/MvcApplication1/Models/IProductRepository.vb
Public Interface IProductRepository
	Function List() As IEnumerable(Of Product)
	Function [Get](ByVal id As Integer) As Product
	Sub Create(ByVal productToCreate As Product)
	Sub Edit(ByVal productToEdit As Product)
	Sub Delete(ByVal productToDelete As Product)
End Interface

File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/MvcApplication1/MvcApplication1/Models/IProductRepository.cs
using System.Collections.Generic;

namespace MvcApplication1.Models
{
    public interface IProductRepository
    {
        IEnumerable<Product> List();
        Product Get(int id);
        void Create(Product productToCreate);
        void Edit(Product productToEdit);
        void Delete(Product productToDelete);
    }
}
